packages/client-utils/test/sequence-manager.test.ts

```diff
@@ -0,0 +1,108 @@
+import test from 'ava';
+import {
+  makeSequenceManager,
+  type AccountResponse,
+} from '../src/sequence-manager.js';
+
+const createMockFetchAccountInfo = () => {
+  let mockSequence = '100';
+  const mockAccountNumber = '377';
+  let callCount = 0;
+
+  const fetch = async (address: string): Promise<AccountResponse> => {
+    callCount += 1;
+    return {
+      account: {
+        '@type': '/cosmos.auth.v1beta1.BaseAccount',
+        address,
+        account_number: mockAccountNumber,
+        sequence: mockSequence,
+      },
+    };
+  };
+
+  return {
+    fetch,
+    updateSequence: (newSequence: string) => {
+      mockSequence = newSequence;
+    },
+    getCallCount: () => callCount,
+  };
+};
```

Comment:
Nit: Rather than hard-coding the mocked data, it should be fully configurable for better locality in the test functions.
```suggestion
const createMockFetchAccountInfo = (
  accountNumber: `${bigint}`,
  sequenceNumber: `${bigint}`,
) => {
  let callCount = 0;

  const fetch = async (address: string): Promise<AccountResponse> => {
    callCount += 1;
    return {
      account: {
        '@type': '/cosmos.auth.v1beta1.BaseAccount',
        address,
        account_number: accountNumber,
        sequence: sequenceNumber,
      },
    };
  };

  return {
    fetch,
    setSequenceNumber: (newSequenceNumber: `${bigint}`) => {
      sequenceNumber = newSequenceNumber;
    },
    getCallCount: () => callCount,
  };
};
```